resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -387,7 +446,7 @@ private[spark] class Client(
         .withLabels(driverKubernetesSelectors)
         .endMetadata()
       .withNewSpec()
-        .withType("NodePort")
+        .withType(if (useIngress) "ClusterIP" else "NodePort")
```

Comment:
Yes, nodeport would definitely capture some use cases. But folks may have on-prem ingress controllers which use ClusterIP. I'm thinking we should support both of these.